components/page.go

```diff
@@ -30,6 +30,22 @@ type Page struct {
 	Layout Layout
 }
 
+type PageOpts func(p *Page)
+
+func (p *Page) SetPageOptions(opts ...PageOpts) *Page {
```

Comment:
Also i think you can add the `opts` in the Page's constructor i wont break anything